all/projectAlcazar/src/sys/timer.c

```diff
@@ -51,6 +51,107 @@ unsigned int getTimerValue(volatile Timer_Type* timer)
+	 * We will use PWM mode 1, edge-aligned, upcounting configuration
+	 *
+	 * Also you should study the example in section
+	 * A.9.8 Edge-aligned PWM configuration example
+	 * keep in mind that in that example CPU is clocked at 48Mhz
+	 *
+	 * We will use the same steps as in the example numbered (x)
+	 *
+	 */
+
+	/*The prescaler can divide the counter clock frequency by any factor between 1 and 65536.
+	 * It is based on a 16-bit counter controlled through a 16-bit register
+	 * (in the TIMx_PSC register). It can be changed on the fly as this control register
+	 * is buffered. The new prescaler ratio is taken into account at the next update event.*/
+	//The counter clock frequency (CK_CNT) is equal to fCK_CPU / (PSC[15:0] + 1).
+
+	// (1) set the prescaler
+	timer->PSC = pwmPeriodPrescaler;
+
+	// (2) set the automatic reload value
+	timer->ARR = pwmPeriodCounter;
+
+	//Using CCRx registers we control the PWM dutycycle for each of the three channels
+	//For PWM mode 1 - In upcounting, channel x is active (HIGH or 1)
+ 	// 	 as long as TIMx_CNT<TIMx_CCRx else inactive (LOW or 0).
+
+	//Because the timer counter goes up only until it reaches ARR register
+	//by setting CCRx register from 0 to ARR we get a duty cycle from 0% to 100%
+
+	// (3) set CCRx registers for the two channels
+	//this sets the duty cycle to 0%, permanent low signal (0) for all 2 channels
+	timer->CCR3 = timer->CCR4 = 0;
+
+	//For setting the output compare mode in CCMR register,
+	//must first clear the CCxE bits in CCER
+	//>> Note: CC2S bits are writable only when the channel is OFF (CCxE = �0� in TIMx_CCER).
+	timer->CCER = 0;
+
+	//see section 17.4.7. We set the 3 channels in output compare mode.
+	//First 2 channels are set using CCMR1 and the third channel is set using CCMR2 register
+	//Bits 0-7 control first channel 8-15 second channel and bits 0-7 of CCMR2 third channel
+	/*
+	 CCxS bits will be outputs. We want to output PWM signal. To change the dutycycle
+	 we will use CCRx registers to set the compare value for each channel.
+	 To be able to set CCRx registers anytime we must enable preloading
+	 (The TIMx_CCRx registers can be programmed with or without preload
+	 registers using the OCxPE bit in the TIMx_CCMRx register.)
+	 Bits 0:1 	CCxS 	=> 00 channel is output
+	 Bit  2   	OCxFE 	=> 0 normal behavior
+	 Bit  3		OCxPE 	=> 1 should be 1 for PWM see 17.3.10
+	 Bits 6:4	OCxM	=> 110: (=0x6) PWM mode 1 - In upcounting, channel 1 is active (HIGH or 1)
+	 	 	 	 	 	 	 	 as long as TIMx_CNT<TIMx_CCR1 else inactive (LOW or 0).
+	 Bit  7		OCxCE	=> 0: OC1Ref is not affected by the ETRF Input
+	 */
+
+	//we have the same configuration for all 3 channels so we can use
+	//the same expression and shift it by 8 to the left for channel 2
+	//so that we end up modifying bits 15 - 8
+
+	// (4) set the capture and compare mode registers for our two channels
+	timer->CCMR1 =  (TIM_MODE_PWM1 << TIM_OCxM | 1 << TIM_OCxPE) << 8 |  //configuration for channel 4
```

Comment:
CCMR2 should be used that controls channels 3 & 4 CCMR1 controls channels 1 and 2